src/cpp/fastdds/publisher/DataWriterImpl.cpp

```diff
@@ -297,6 +297,15 @@ ReturnCode_t DataWriterImpl::enable()
         return ReturnCode_t::RETCODE_ERROR;
     }
 
+    history_.unacknowledged_sample_removed_functor([&](
```

Comment:
See my previous comment regarding injecting the functor when constructing the history.

Reply:
Done in c8cde8cad